papermill/iorw.py

```diff
@@ -227,7 +225,7 @@ def pretty_path(cls, path):
 
 def read_yaml_file(path):
     """Reads a YAML file from the location specified at 'path'."""
-    return yaml.load(papermill_io.read(path))
+    return yaml.load(papermill_io.read(path, ['.json', '.yaml', '.yml']))
```

Comment:
Is this the one change that is needed to fix the bug? If no then it would be good if you could tell me how the other functions get involved. I feel like I am missing something as `read_yaml_file()` wouldn't have raised the warning that was reported.

Reply:
Ok, five minutes later I now also see `papermill_io.read` in the call here ... 👀. All good.